### PR TITLE
Call location.loadById in RouteStop

### PR DESCRIPTION
**Pull request by radkaag**

### CODE REVIEW

**Comment by mrtnzlml** [comment]
@radkaag The tests are still failing. Please do not let your PRs be stale... 🙂

File: config/application.js [hunk 1]
```diff
@@ -21,11 +21,12 @@ export default {
     login: 'https://auth.skypicker.com/v1/user.login',
     identity: 'https://auth.skypicker.com/v1/user.get',
     airlines: 'https://api.skypicker.com/airlines?v=2',
-    allLocations: (queryParameters: Object) =>
-      queryWithParameters(
+    allLocations: (queryParameters: Object) => {
```
Comment (michalsanger) [inline config/application.js]: Why this change?

File: src/flight/queries/__tests__/AllFlights.locale.test.js [hunk 2]
```diff
@@ -38,17 +38,34 @@ beforeEach(() => {
...
+  RestApiMock.onGet(
+    config.restApiEndpoint.allLocations({
+      term: 'Praha',
+      locale: 'cs-CZ',
+    }),
+  ).replyWithData(PragueCsCzDataset);
+
+  RestApiMock.onGet(
+    config.restApiEndpoint.allLocations({
+      term: 'MEX',
```
Comment (michalsanger) [inline src/flight/queries/__tests__/AllFlights.locale.test.js]: We agreed this is a bug, `MEX` should be fetch via id not via term URL. Let's accept this hack not to block the PR. But please create issue for this and add comment here:
```js
// it's a hack, will be fixed here {ISSUE_URL}
```

File: src/flight/types/outputs/RouteStop.js [hunk 3]
```diff
@@ -20,13 +20,9 @@ export default new GraphQLObjectType({
...
       ): Promise<LocationType> => {
-        let queryOptions = options.getOptions(path) || {};
-
-        if (!queryOptions.locale) {
-          queryOptions = { ...queryOptions, locale };
-        }
-
-        return dataLoader.location.load(where.code, queryOptions);
+        const queryOptions = options.getOptions(path) || {};
+        const useLocale = queryOptions.locale || locale;
```
Comment (michalsanger) [inline src/flight/types/outputs/RouteStop.js]: Strange naming, let's try `selectedLocale`

Comment (radkaag) [reply src/flight/types/outputs/RouteStop.js]: Why then everywhere in the code is used `useLocale`?

File: src/flight/queries/__tests__/AllFlights.locale.test.js (hunk 2 above)
Comment (michalsanger) [reply src/flight/queries/__tests__/AllFlights.locale.test.js]: OMG yes, I finally dig into this problem. @radkaag I'll explain that f2f, it's not a bug :-)

The `should return flight in czech language with location fallback` test has 2 mocks for flights result. First for "Praha > MEX" no flight is returned, so the fallback is about "let's try to fetch location detail with search-by-term REST API and call the flights REST API again". That's the second mock for "prague_cz > MEX". 

Our mistake was to think we always fetch location detail for flight search inputs and then call flights REST API. But what we really do is call the flights REST API first and if no results are returned, try to normalize the locations.

Comment (mrtnzlml) [reply src/flight/queries/__tests__/AllFlights.locale.test.js]: Exactly. We can however 1) improve this flow to be more obvious and predictable or 2) describe it better and write better tests for this (more readable).

Comment (michalsanger) [reply src/flight/queries/__tests__/AllFlights.locale.test.js]: I'd start with putting each test into own file :-)

File: src/flight/types/outputs/RouteStop.js (hunk 3 above)
Comment (radkaag) [reply src/flight/types/outputs/RouteStop.js]: I rename all files.

File: src/flight/queries/__tests__/AllFlights.locale.test.js [hunk 4]
```diff
@@ -38,17 +38,35 @@ beforeEach(() => {
...
+  ).replyWithData(MexCsCzDataset);
+
+  RestApiMock.onGet(
+    config.restApiEndpoint.allLocations({
+      term: 'Praha',
+      locale: 'cs-CZ',
+    }),
+  ).replyWithData(PragueCsCzDataset);
+
+  // it's a hack, will be fixed here {https://github.com/kiwicom/graphql/issues/141}
```
Comment (mrtnzlml) [inline src/flight/queries/__tests__/AllFlights.locale.test.js]: Please remove this comment. See: https://github.com/kiwicom/graphql/pull/133/#discussion_r187158477

File: src/flight/queries/__tests__/AllFlights.locale.test.js (hunk 2 above)
Comment (radkaag) [reply src/flight/queries/__tests__/AllFlights.locale.test.js]: Take look at splitting the code into two files.

File: src/flight/types/outputs/RouteStop.js [hunk 5]
```diff
@@ -20,13 +20,9 @@ export default new GraphQLObjectType({
...
       ): Promise<LocationType> => {
-        let queryOptions = options.getOptions(path) || {};
-
-        if (!queryOptions.locale) {
-          queryOptions = { ...queryOptions, locale };
-        }
-
-        return dataLoader.location.load(where.code, queryOptions);
+        const queryOptions = options.getOptions(path) || {};
+        const selectedLocale = queryOptions.locale || locale;
```
Comment (michalsanger) [inline src/flight/types/outputs/RouteStop.js]: ❤️
